index.js

```diff
@@ -42,13 +47,27 @@ const makeEntityMapKeys = (sys) =>
  * @return {String}
  */
 const lookupInEntityMap = (entityMap, linkData) => {
-  const { entryId, linkType, spaceId } = linkData
-  if (spaceId) {
-    return entityMap.get(`${spaceId}!${linkType}!${entryId}`)
+  const { entryId, linkType, spaceId, environmentId } = linkData
+
+  if (spaceId && environmentId) {
+    return entityMap.get(`${spaceId}!${environmentId}!${linkType}!${entryId}`)
   }
+
   return entityMap.get(`${linkType}!${entryId}`)
 }
 
+const getIdsFromUrn = (urn) => {
+  const regExp =
```

Comment:
Could you also add a `TODO` comment to update this once the `resource-names` package becomes public so we can avoid this code duplication for the regex, and be able to know where/what part of the code to update